Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt

```diff
@@ -153,24 +153,32 @@ private fun MapViewContainer(
     latitude: String,
     longitude: String
 ) {
-    var zoom by rememberSaveable { mutableStateOf(InitialZoom) }
-    val coroutineScope = rememberCoroutineScope()
+    var mapInitialized by remember(map) { mutableStateOf(false) }
+    LaunchedEffect(map, mapInitialized) {
+        if (!mapInitialized) {
+            val googleMap = map.awaitMap()
+            val position = LatLng(latitude.toDouble(), longitude.toDouble())
+            googleMap.addMarker {
+                position(position)
+            }
+            googleMap.moveCamera(CameraUpdateFactory.newLatLng(position))
+            mapInitialized = true
+        }
+    }
 
+    var zoom by remember(map) { mutableStateOf(InitialZoom) }
```

Comment:
This should probably still be `rememberSaveable`?

Reply:
Makes sense. Thanks!